utils/network.go

```diff
@@ -340,3 +345,41 @@ func writeProcSys(path, value string) error {
 	}
 	return err
 }
+
+// Generate a mac address from the passed in string
+func genLocalMacAddr(src string) net.HardwareAddr {
+	// The 2nd bit set of the first octet makes this MAC address a "locally administered address"
+	// https://en.wikipedia.org/wiki/MAC_address
+	macString := "02"
+
+	srcBytes := strings.Split(src, "")
+	srcLen := len(srcBytes)
+	for i, j := 0, 0; i < 10; i++ {
+		if i%2 == 0 {
+			macString += ":"
+		}
+
+		var m string
+		m = ""
+		// Go through the next bytes until a hex one is found
+		for ; j < srcLen && m == ""; j++ {
+			// Only use the byte if the decode succeeds
+			if _, err := hex.DecodeString("0" + srcBytes[j]); err == nil {
+				m = srcBytes[j]
+			} else {
+				logrus.Infof("Convert %s failed with %v", srcBytes[j], err)
+			}
+		}
+		if m == "" {
+			m = "0"
+		}
+		macString += m
```

Comment:
Since you're appending rather than prepending, the "02" above is going to be at the beginning of the mac address which doesn't work for locally administered address since it'll be the MSB rather than the LSB.

Reply:
The way I read the wiki article I linked I think setting that bit is correct. It says `Universally administered and locally administered addresses are distinguished by setting the second-least-significant bit of the first octet of the address`  and even gives the example of `06-00-00-00-00-00` as being a "locally administered address".

Reply:
You're right, my mistake. Its the 2nd bit of the first (not the 6th) octet which is what I (mis)read.